### PR TITLE
Moving GetRepo logic to the non-UI thread

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
The repo tool uses the UI thread to obtain all repositories for an account.

The long running code is moved to a background thread.

## References and relevant issues
#479 
## Validation steps performed
Manually ran.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

Movie!
GetReposOnADifferentThread

### CODE REVIEW

**Comment by dhoehna** [comment]
Closes #479

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs [hunk 1]
```diff
@@ -538,17 +539,20 @@ public void AddRepositoryViaUri(string url, string cloneLocation)
...
+            var loggedInDeveloper = _providers.GetAllLoggedInAccounts(repositoryProvider).FirstOrDefault(x => x.LoginId() == loginId);
 
-        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetRepos_Event", LogLevel.Measure, new RepoToolEvent("GettingAllRepos"));
-        _repositoriesForAccount = _providers.GetAllRepositories(repositoryProvider, loggedInDeveloper);
+            TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetRepos_Event", LogLevel.Measure, new RepoToolEvent("GettingAllRepos"));
+            _repositoriesForAccount = _providers.GetAllRepositories(repositoryProvider, loggedInDeveloper);
+        });
 
-        Repositories = new ObservableCollection<RepoViewListItem>(OrderRepos(_repositoriesForAccount));
+        Application.Current.GetService<WindowEx>().DispatcherQueue.TryEnqueue(() => { Repositories = new ObservableCollection<RepoViewListItem>(OrderRepos(_repositoriesForAccount)); });
```
Comment (AmelBawa-msft) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:555]: Can we omit the dispatcher queue for setting the observable property from the caller thread (assuming it's the UI thread, e.g. method bound to a relay command or an event handler)?

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs [hunk 2]
```diff
@@ -538,17 +539,20 @@ public void AddRepositoryViaUri(string url, string cloneLocation)
     /// </summary>
     /// <param name="repositoryProvider">The provider.  This should match the display name of the plugin</param>
     /// <param name="loginId">The login Id to get the repositories for</param>
-    public IEnumerable<RepoViewListItem> GetRepositories(string repositoryProvider, string loginId)
+    public async Task<IEnumerable<RepoViewListItem>> GetRepositories(string repositoryProvider, string loginId)
```
Comment (AmelBawa-msft) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:542]:
```suggestion
    public async Task<IEnumerable<RepoViewListItem>> GetRepositoriesAsync(string repositoryProvider, string loginId)

```

**Review by AmelBawa-msft — APPROVED** [review]

**Review by krschau — APPROVED** [review]
It might be long enough a pause that it's worth sticking a ProgressRing in there while you fetch the repos.

**Pull request merged**